### PR TITLE
Adding Nucleo_f401re support

### PR DESCRIPTION
**Pull request by FredPraca**
Here we go once again :smile:

### CODE REVIEW

File: arm/stm32/stm32f401/memory-map.ld [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *                                                                          *
+ *                         GNAT COMPILER COMPONENTS                         *
+ *                                                                          *
+ *                                  A R M                                   *
+ *                                                                          *
+ *                            Linker Script File                            *
+ *                                                                          *
+ *      Copyright (C) 1999-2002 Universidad Politecnica de Madrid           *
+ *             Copyright (C) 2003-2006 The European Space Agency            *
+ *                   Copyright (C) 2003-2015 AdaCore                        *
```
Comment (Fabien-Chouteau) [inline arm/stm32/stm32f401/memory-map.ld:11]: This should be `2003-2020`

File: arm/stm32/stm32f401/s-bbmcpa.adb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+------------------------------------------------------------------------------
+--                                                                          --
+--                  GNAT RUN-TIME LIBRARY (GNARL) COMPONENTS                --
+--                                                                          --
+--              S Y S T E M . B B . M C U _ P A R A M E T E R S             --
+--                                                                          --
+--                                  B o d y                                 --
+--                                                                          --
+--                      Copyright (C) 2016, AdaCore                         --
```
Comment (Fabien-Chouteau) [inline arm/stm32/stm32f401/s-bbmcpa.adb:9]: This should be `2020`

File: arm/stm32/stm32f401/s-bbmcpa.ads [hunk 3]
```diff
@@ -0,0 +1,54 @@
+------------------------------------------------------------------------------
+--                                                                          --
+--                  GNAT RUN-TIME LIBRARY (GNARL) COMPONENTS                --
+--                                                                          --
+--              S Y S T E M . B B . M C U _ P A R A M E T E R S             --
+--                                                                          --
+--                                  S p e c                                 --
+--                                                                          --
+--                      Copyright (C) 2016, AdaCore                         --
```
Comment (Fabien-Chouteau) [inline arm/stm32/stm32f401/s-bbmcpa.ads:9]: This should be `2020`

File: arm/stm32/stm32f401/s-stm32.adb [hunk 4]
```diff
@@ -0,0 +1,132 @@
+------------------------------------------------------------------------------
+--                                                                          --
+--                         GNAT RUN-TIME COMPONENTS                         --
+--                                                                          --
+--          Copyright (C) 2012-2016, Free Software Foundation, Inc.         --
```
Comment (Fabien-Chouteau) [inline arm/stm32/stm32f401/s-stm32.adb:5]: This should be `2012-2020`

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

File: arm/stm32/stm32f401/memory-map.ld (hunk 1 above)
Comment (FredPraca) [reply arm/stm32/stm32f401/memory-map.ld]: Done

**Comment by Fabien-Chouteau** [comment]
Thank you @FredPraca  we will review the code and get back to you as soon as possible.

**Comment by Fabien-Chouteau** [comment]
The PR is merged, thank you @FredPraca !

**Pull request merged**